pkg/middleware/validation.go

```diff
@@ -45,31 +49,50 @@ func NewValidationMiddleware(validatorHook string) mux.MiddlewareFunc {
 	}
 }
 
+func maybeLogValidationReason(context context.Context, message string, mod string, version string) {
+	if len(message) > 0 {
+		entry := log.EntryFromContext(context)
+		entry.Warnf("error validating %s@%s %s", mod, version, message)
+	}
+}
+
 type validationParams struct {
 	Module  string
 	Version string
 }
 
-func validate(hook, mod, ver string) (bool, error) {
+type ValidationResponse struct {
+	Valid   bool
+	Message []byte
+}
+
+func validate(hook, mod, ver string) (ValidationResponse, error) {
 	const op errors.Op = "actions.validate"
 
 	toVal := &validationParams{mod, ver}
 	jsonVal, err := json.Marshal(toVal)
 	if err != nil {
-		return false, errors.E(op, err)
+		return ValidationResponse{Valid: false}, errors.E(op, err)
 	}
 
 	resp, err := http.Post(hook, "application/json", bytes.NewBuffer(jsonVal))
 	if err != nil {
-		return false, errors.E(op, err)
+		return ValidationResponse{Valid: false}, errors.E(op, err)
 	}
 
-	switch {
-	case resp.StatusCode == http.StatusOK:
-		return true, nil
-	case resp.StatusCode == http.StatusForbidden:
-		return false, nil
+	switch resp.StatusCode {
+	case http.StatusOK:
+		return maybeReadResponseReason(resp), nil
+	case http.StatusForbidden:
+		return maybeReadResponseReason(resp), nil
 	default:
-		return false, errors.E(op, "Unexpected status code ", resp.StatusCode)
+		return ValidationResponse{Valid: false}, errors.E(op, "Unexpected status code ", resp.StatusCode)
 	}
 }
+
+func maybeReadResponseReason(resp *http.Response) ValidationResponse {
```

Comment:
💡 rename to something like `validationResponseFromRequest`?or something similar

Reply:
I like this change as well

Reply:
https://github.com/gomods/athens/pull/1517/commits/f578bf9872c7ff65b8682e8734320725d8154ebf